apps/ios/LiveVideo/LiveVideo/Views/StreamConfigFlow/EnterStreamNameView.swift

```diff
@@ -8,59 +8,75 @@ struct EnterStreamNameView: View {
             )
         }
     }
 
     var body: some View {
         NavigationView {
-            FormStack {
-                Text(viewModel.tip)
-                    .modifier(TipStyle())
-
-                TextField("Event name", text: $streamName)
-                    .textFieldStyle(FormTextFieldStyle())
-                    .autocapitalization(.none)
-                    .disableAutocorrection(true)
-
-                VStack {
-                    NavigationLink(destination: SelectRoleView(), isActive: $isShowingSelectRole) {
-                        EmptyView()
+            Form {
+                Section(footer: Text(viewModel.tip)
+                ) {
+                    TextField("Event name", text: $streamName)
+                        .autocapitalization(.none)
+                        .disableAutocorrection(true)
+                }
+                
+                if viewModel.shouldShowRecordOption {
+                    Section(header: Text("Recording")) {
+                        Toggle("Record event", isOn: $shouldRecord)
                     }
-
-                    Button("Continue") {
-                        flowModel.parameters.streamName = streamName
-                        flowModel.parameters.userIdentity = authManager.userIdentity
-                        
-                        if viewModel.shouldSelectRole {
-                            isShowingSelectRole = true
-                        } else {
-                            flowModel.isShowing = false
+                }
+                
+                Section(
+                    footer:
+                        VStack {
+                            NavigationLink(destination: SelectRoleView(), isActive: $isShowingSelectRole) {
+                                EmptyView()
+                            }
+                            
+                            Button("Continue") {
+                                flowModel.parameters.streamName = streamName
+                                flowModel.parameters.userIdentity = authManager.userIdentity
+                                flowModel.parameters.shouldRecord = shouldRecord
+                                
+                                if viewModel.shouldSelectRole {
+                                    isShowingSelectRole = true
+                                } else {
+                                    flowModel.isShowing = false
+                                }
+                            }
+                            .buttonStyle(PrimaryButtonStyle(isEnabled: !streamName.isEmpty))
+                            .disabled(streamName.isEmpty)
+                            .padding(.horizontal, -20)
                         }
-                    }
-                    .buttonStyle(PrimaryButtonStyle(isEnabled: !streamName.isEmpty))
-                    .disabled(streamName.isEmpty)
+                ) {
+                    /// Using the footer to have more control over style
```

Comment:
Is this intended ?

Reply:
Yes this is intended. We get lots of benefits from using a `SwiftUI.Form` on input screens like this. A lot of the styling and layout is done automatically. However if we want to display a big blue button on this screen we can't just add it to the section body because the body will apply extra styling we don't want. Headers and footers don't apply the extra styling and so as a trick we can create a section with empty body and put the button in the footer.

Why not just add this button to the footer of the previous section? Well the sections on this screen are dynamic depending if the user is host or not. And so it would get complex.

It would be nice if there was another way but kind of a small price to pay since we get so many benefits from using `Form` on this screen.